src/applications/disability-benefits/2346/components/Batteries.jsx

```diff
@@ -35,41 +35,29 @@ class Batteries extends Component {
   };
 
   render() {
-    const { supplies, selectedProducts } = this.props;
+    const { supplies, selectedProducts, eligibility } = this.props;
     const currentDate = moment();
     const batterySupplies = supplies.filter(
       batterySupply => batterySupply.productGroup === HEARING_AID_BATTERIES,
     );
-    const areBatterySuppliesIneligible = batterySupplies.every(
-      batterySupply => batterySupply.availableForReorder === false,
-    );
+    const areBatterySuppliesEligible = eligibility.batteries;
+    const haveBatteriesBeenOrderedInLastFiveMonths =
+      batterySupplies.length > 0
+        ? batterySupplies.every(
+            battery => currentDate.diff(battery.lastOrderDate, 'months') <= 5,
+          )
+        : false;
```

Comment:
You can actually rewrite this to 
```
batterySupplies.every(battery => currentDate.diff(battery.lastOrderDate, 'months') <= 5).length > 0
``` 
to remove the ternary

Reply:
> You can actually rewrite this to
> 
> ```
> batterySupplies.every(battery => currentDate.diff(battery.lastOrderDate, 'months') <= 5).length > 0
> ```
> 
> to remove the ternary

On user 1, i'm getting `false` for its value when it should be `true`. The `every` function is returning `true` as it should.  I got rid of the length comparison to 0 and just left length and got `undefined`.  I think it's trying to compare the length of a boolean which is why it comes out `undefined`.

Reply:
Still, you can run array prototype methods on an empty array - you shoudln't need a length comparison and thus can remove the ternary

Reply:
I could add a check prior to `every`. Something like:

`batterySupplies.length > 0 &&
      batterySupplies.every(
        battery => currentDate.diff(battery.lastOrderDate, 'months') <= 5,
      );`